Worn Out/Worn Out/Controllers/HealthKitController.swift

```diff
@@ -22,11 +90,7 @@ class HealthKitController {
                                         limit: HKObjectQueryNoLimit,
                                         sortDescriptors: [sortDescriptor]) { (query, results, error) in
                                             if let results = results {
-                                                print("Retrieved the following workouts")
-                                                for sample in results {
-                                                    let workout = sample
-                                                    print("\(workout)")
-                                                }
+                                                completion(results)
```

Comment:
I’m not sure if this API calls back on the main thread or not, but you might want to call your completion handler in DispatchQueue.main.async { } if it does.

Reply:
that makes sense!

Reply:
*if it doesn’t call back on the main queue (otherwise you’ll have threading issues)